source/opt/function.h

```diff
@@ -73,6 +77,14 @@ class Function {
   // Returns function's return type id
   inline uint32_t type_id() const { return def_inst_->type_id(); }
 
+  // Returns the basic block container for this function.
+  std::vector<std::unique_ptr<BasicBlock>>* GetBlocks() { return &blocks_; }
```

Comment:
I did that because I needed access to the container to allow the a generic deletion procedure (one that could work with std::list : https://github.com/KhronosGroup/SPIRV-Tools/pull/1292#discussion_r168379160)

I will move to a vector so this will no longer be necessary.
FYI: I made to tried a more generic UPtrVectorIterator, but VS2013 is not able to compile it.